service/history/tasks/workflow_task_timer.go

```diff
@@ -53,6 +53,7 @@ type (
 		ScheduleAttempt     int32
 		TimeoutType         enumspb.TimeoutType
 		Version             int64
+		Speculative         bool
```

Comment:
I was going back and forth on the name: `Speculative` vs `Memory`. Happy to discuss.